src/batch/src/executor/join/nested_loop_join.rs

```diff
@@ -110,42 +76,72 @@ impl Executor for NestedLoopJoinExecutor {
 
 impl NestedLoopJoinExecutor {
     #[try_stream(boxed, ok = DataChunk, error = RwError)]
-    async fn do_execute(mut self: Box<Self>) {
-        let mut state = NestedLoopJoinState::Build;
-        loop {
-            match state {
-                NestedLoopJoinState::Build => {
-                    self.build_table.load_data().await?;
-                    state = NestedLoopJoinState::FirstProbe;
-                }
-                NestedLoopJoinState::FirstProbe => {
-                    let ret = self.probe(true, &mut state).await?;
-                    if let Some(data_chunk) = ret {
-                        yield data_chunk.reorder_columns(&self.output_indices);
-                    }
-                }
-                NestedLoopJoinState::Probe => {
-                    let ret = self.probe(false, &mut state).await?;
-                    if let Some(data_chunk) = ret {
-                        yield data_chunk.reorder_columns(&self.output_indices);
-                    }
-                }
+    async fn do_execute(self: Box<Self>) {
+        let left_data_types = self.left_child.schema().data_types();
+        let data_types = self.original_schema.data_types();
+
+        let mut chunk_builder = DataChunkBuilder::with_default_size(data_types);
+
+        // Cache the outputs of left child
+        let left = self.left_child.execute().try_collect().await?;
+
+        // Get the joined stream
+        let stream = match self.join_type {
+            JoinType::Inner => Self::do_inner_join,
+            JoinType::LeftOuter => Self::do_left_outer_join,
+            JoinType::LeftSemi => Self::do_left_semi_anti_join::<false>,
+            JoinType::LeftAnti => Self::do_left_semi_anti_join::<true>,
+            JoinType::RightOuter => Self::do_right_outer_join,
+            JoinType::RightSemi => Self::do_right_semi_anti_join::<false>,
+            JoinType::RightAnti => Self::do_right_semi_anti_join::<true>,
+            JoinType::FullOuter => todo!(),
```

Comment:
We left it `FullOuter` as `todo!()` because original version didn't implement this?

Reply:
> We left it `FullOuter` as `todo!()` because original version didn't implement this?

Yep. But it seems trivial to implement this. Maybe finish it in later PRs.

Reply:
Indeed. We can add a issue to impl if we want.

Previous cuz PG do not support FullOuter. They achieve it by Union(RightOuter, LeftOuter)

Reply:
+1 for add it in later PR.